cli/command/service/opts.go

```diff
@@ -190,13 +196,15 @@ type updateOptions struct {
 	maxFailureRatio floatValue
 }
 
-func (opts updateOptions) config() *swarm.UpdateConfig {
+func (opts updateOptions) defaults(defaultUpdateConfig *api.UpdateConfig) *swarm.UpdateConfig {
```

Comment:
This is passed either `defaults.Service.Update` or `defaults.Rollback.Update`. I agree that it's strange as a method on `updateOpts`. I'll change it to a function.

Any preference on whether it should be a single function that takes a `*api.UpdateConfig` (like this method), or separate functions that call a shared implementation with either the `Update` or `Rollback` default values?